spec-tests/src/test/java/org/candlepin/spec/jobs/JobStatusSpecTest.java

```diff
@@ -0,0 +1,480 @@
+        // pool.setUpstreamPoolId(StringUtil.random("upstream"));
+        // pool.setQuantity(4L);
+        pool.setStartDate(Instant.now().atOffset(ZoneOffset.UTC));
+        pool.setEndDate(Instant.now().plus(365, ChronoUnit.DAYS).atOffset(ZoneOffset.UTC));
+        pool = ownerApi.createPool(owner.getKey(), pool);
+    }
+
+    @Test
+    @DisplayName("should find an empty list if the owner key is wrong")
+    public void shouldFindEmptyListOwnerNotExist() throws Exception {
+        List<AsyncJobStatusDTO> jobs = jobsClient.listMatchingJobStatusForOrg("totally-made-up",
+            null, null);
+        assertThat(jobs).isEmpty();
+    }
+
+    @Test
+    @DisplayName("should cancel a job")
+    public void shouldCancelAJob() throws Exception {
+        jobsClient.setSchedulerStatus(false);
+        try {
+            AsyncJobStatusDTO jobStatus = ownerApi.healEntire(owner.getKey());
+            // make sure we see a job waiting to go
+            List<AsyncJobStatusDTO> statuses = jobsClient.listMatchingJobStatusForOrg(owner.getKey(),
+                Set.of(jobStatus.getId()), "CREATED");
+            assertThat(statuses).hasSize(1);
+
+            jobsClient.cancelJob(jobStatus.getId());
+            // make sure we see a job canceled
+            statuses = jobsClient.listMatchingJobStatusForOrg(owner.getKey(),
+                Set.of(jobStatus.getId()), "CANCELED");
+            assertThat(statuses).hasSize(1);
+
+            jobsClient.setSchedulerStatus(true);
+            sleep(1); // let the job queue drain..
+            // make sure job didn't flip to FINISHED
+            statuses = jobsClient.listMatchingJobStatusForOrg(owner.getKey(),
+                Set.of(jobStatus.getId()), "CANCELED");
+            assertThat(statuses).hasSize(1);
+        }
+        finally {
+            jobsClient.setSchedulerStatus(true);
+        }
+    }
+
+    @Test
+    @DisplayName("should contain the system id for async binds")
+    public void shouldContainSystemIdForAsync() throws Exception {
+        ownerApi.healEntire(owner.getKey());
+        ConsumerDTO consumer = client.consumers().register(Consumers.random(owner));
+        // need a consumer api client
+        ApiClient consumerClient = ApiClients.trustedConsumer(consumer.getUuid());
+        consumerApi = consumerClient.consumers();
+        AsyncJobStatusDTO bindStatus = AsyncJobStatusDTO.fromJson(consumerApi.bind(consumer.getUuid(),
+            null, List.of(product.getId()), null, null, null, true, null, null));
+        assertEquals(consumer.getUuid(), bindStatus.getPrincipal());
+        jobsClient.waitForJobToComplete(bindStatus.getId(), 15);
+    }
+
+    @Test
+    @DisplayName("should allow admin to view any job status")
+    public void shouldAllowAdminToViewJobStatus() throws Exception {
+        AsyncJobStatusDTO jobStatus = ownerApi.healEntire(owner.getKey());
+        jobsClient.waitForJobToComplete(jobStatus.getId(), 15);
```

Comment:
Should this be 15 seconds or 15 milliseconds?

It looks like this also a question for some of the other tests in this file using the waitForJobToComplete method.